### PR TITLE
GG-28222 CPP Thin: Use random address order

### PR DESCRIPTION
**Pull request by isapego**

### CODE REVIEW

File: modules/platforms/cpp/network/os/win/src/network/tcp_socket_client.cpp [hunk 1]
```diff
@@ -28,6 +30,23 @@ namespace ignite
...
+        {
+            std::vector<addrinfo*> res;
+
+            for (addrinfo *it = addrsIn; it != NULL; it = it->ai_next)
+                res.push_back(it);
+
+            std::random_shuffle(res.begin(), res.end());
+
+            return res;
+        }
```
Comment (ptupitsyn) [inline modules/platforms/cpp/network/os/win/src/network/tcp_socket_client.cpp:48]: This looks exactly like the code above for Linux, can we deduplicate?

**Review by ptupitsyn — APPROVED** [review]
Looks good to me in general, please see one comment.

File: modules/platforms/cpp/network/os/win/src/network/tcp_socket_client.cpp (hunk 1 above)
Comment (isapego) [reply modules/platforms/cpp/network/os/win/src/network/tcp_socket_client.cpp:48]: Well in theory, we can, using templates. The hard part here is that despite of the same name `addrinfo`s in Windows and Linux are two different classes, so we can not just make a simple function. I'll think if I can improve this part without over-complicating the code though.

Comment (isapego) [reply modules/platforms/cpp/network/os/win/src/network/tcp_socket_client.cpp]: Fixed. Check out new version.

Comment (ptupitsyn) [reply modules/platforms/cpp/network/os/win/src/network/tcp_socket_client.cpp]: Looks good, thanks!

**Pull request merged**